jstests/aggregation/bugs/server7695_isodates.js

```diff
@@ -0,0 +1,284 @@
+    var years = {
+        common: [
+            1900, // Starting and ending on Monday (special).
+            2002, // Starting and ending on Tuesday.
+            2014, // Starting and ending on Wednesday.
+            2015, // Starting and ending on Thursday.
+            2010, // Starting and ending on Friday.
+            2011, // Starting and ending on Saturday.
+            2006, // Starting and ending on Sunday.
+        ],
+        leap: [
+            1996, // Starting on Monday, ending on Tuesday.
+            2008, // Starting on Tuesday, ending on Wednesday.
+            1992, // Starting on Wednesday, ending on Thursday.
+            2004, // Starting on Thursday, ending on Friday.
+            2016, // Starting on Friday, ending on Saturday.
+            2000, // Starting on Saturday, ending on Sunday (special).
+            2012, // Starting on Sunday, ending on Monday.
+        ],
+        commonAfterLeap: [
+            2001, // Starting and ending on Monday.
+            2013, // Starting and ending on Tuesday.
+            1997, // Starting and ending on Wednesday.
+            2009, // Starting and ending on Thursday.
+            1993, // Starting and ending on Friday.
+            2005, // Starting and ending on Saturday.
+            2017, // Starting and ending on Sunday.
+        ],
+    };
+
+    var MONDAY = 1;
+    var TUESDAY = 2;
+    var WEDNESDAY = 3;
+    var THURSDAY = 4;
+    var FRIDAY = 5;
+    var SATURDAY = 6;
+    var SUNDAY = 7;
+
+    function padded(val) {
+        return ("00" + val).slice(-2);
+    }
+    function getNewYear(year) {
+        return new Date("" + year + "-01-01T00:00:00Z");
+    }
+    function getEndOfFirstWeekInYear(year,day) {
+        return new Date("" + year + "-01-"+(padded(7-day+1))+"T23:59:59Z");
+    }
+    function getStartOfSecondWeekInYear(year,day) {
+        return new Date("" + year + "-01-"+(padded(7-day+2))+"T00:00:00Z");
+    }
+    function getBirthday(year) {
+        return new Date("" + year + "-07-05T21:36:00+02:00");
+    }
+    function getEndOfSecondToLastWeekInYear(year,day,type) {
+        if (type === 'leap') {
+            return new Date("" + year + "-12-"+padded(31-day-1)+"T23:59:59Z");
+        } else {
+            return new Date("" + year + "-12-"+padded(31-day)+"T23:59:59Z");
+        }
+    }
+    function getStartOfLastWeekInYear(year,day,type) {
+        if (type === 'leap') {
+            return new Date("" + year + "-12-"+padded(31-day)+"T00:00:00Z");
+        } else {
+            return new Date("" + year + "-12-"+padded(31-day+1)+"T00:00:00Z");
+        }
+    }
+    function getNewYearsEve(year) {
+        return new Date("" + year + "-12-31T23:59:59Z");
+    }
+    function shiftWeekday(day, days) {
+        return ((day - 1 + days) % 7) + 1;
+    }
+
+    ['common', 'leap', 'commonAfterLeap'].forEach(function(type) {
+        years[type].forEach(function(year, day) {
```

Comment:
`day` is not a very descriptive variable name here... it really refers to the index in the array. I'd suggest renaming this variable to `i`, then re-working the comment on line 123 to describe how the index corresponds to the day.